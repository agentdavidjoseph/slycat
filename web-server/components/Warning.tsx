import * as React from "react";


export interface WarningProps {
  warningMessage:Array<string>
}

export default class Warning  extends React.Component<WarningProps> {
  public constructor(props:WarningProps) {
    super(props);
  }

  render () {
    return(
      <div>
        {this.props.warningMessage.map((message, i) => {
          return (
            <div key={i}>
              <label style={{color: 'red', fontSize:'15.5px', textAlign:'center', backgroundColor:'#FFFF99', fontWeight:'bold'}}>
                {message}
              </label>
            </div>
          );})}
        </div>
    );
  }
}