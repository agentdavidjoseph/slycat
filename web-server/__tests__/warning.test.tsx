
import React from 'react';
import Warning, {WarningProps, WarningState} from "components/Warning";
import { mount } from "enzyme";

describe('when loading a warning message',() =>{
  const message = ['test1','test2','test3'];
  const properties: WarningProps = {
    warningMessage:message,
  };
  const render = mount(
    <Warning
      {...properties}
    />
      );
  test('we have expected props on initial load', () => {
    const props = render.props() as WarningProps;
    expect(props).toMatchSnapshot();
  });
  test('we have expected null state  on initial load', () => {
    const props = render.state() as WarningState;
    expect(props).toMatchSnapshot();
  });
  test('we should have a full initialization', () => {
     expect(render).toMatchSnapshot();
  });
  test('we should be able to change the background color', () => {
    render.setProps({ backgroundColor: 'colorfulness'});
    expect(render).toMatchSnapshot();
  });
});

export default undefined