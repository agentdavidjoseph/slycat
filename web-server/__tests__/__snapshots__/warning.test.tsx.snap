// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when loading a warning message we have expected null state  on initial load 1`] = `null`;

exports[`when loading a warning message we have expected props on initial load 1`] = `
Object {
  "warningMessage": Array [
    "test1",
    "test2",
    "test3",
  ],
}
`;

exports[`when loading a warning message we should be able to change the background color 1`] = `
<Warning
  backgroundColor="colorfulness"
  warningMessage={
    Array [
      "test1",
      "test2",
      "test3",
    ]
  }
>
  <div>
    <label
      key="0"
      style={
        Object {
          "backgroundColor": "colorfulness",
          "color": "red",
          "fontSize": "15.5px",
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      test1
    </label>
    <label
      key="1"
      style={
        Object {
          "backgroundColor": "colorfulness",
          "color": "red",
          "fontSize": "15.5px",
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      test2
    </label>
    <label
      key="2"
      style={
        Object {
          "backgroundColor": "colorfulness",
          "color": "red",
          "fontSize": "15.5px",
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      test3
    </label>
  </div>
</Warning>
`;

exports[`when loading a warning message we should have a full initialization 1`] = `
<Warning
  warningMessage={
    Array [
      "test1",
      "test2",
      "test3",
    ]
  }
>
  <div>
    <label
      key="0"
      style={
        Object {
          "backgroundColor": "",
          "color": "red",
          "fontSize": "15.5px",
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      test1
    </label>
    <label
      key="1"
      style={
        Object {
          "backgroundColor": "",
          "color": "red",
          "fontSize": "15.5px",
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      test2
    </label>
    <label
      key="2"
      style={
        Object {
          "backgroundColor": "",
          "color": "red",
          "fontSize": "15.5px",
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      test3
    </label>
  </div>
</Warning>
`;
